### PR TITLE
spgemm unit test: update cusparse macro check

### PR DESCRIPTION
**Pull request by ndellingwood**
Similar fix as in PR #290 for MKL

### CODE REVIEW

File: unit_test/sparse/Test_Sparse_spgemm.hpp [hunk 1]
```diff
@@ -301,7 +301,7 @@ void test_spgemm(lno_t numRows, size_type nnz, lno_t bandwidth, lno_t row_size_v
     case SPGEMM_CUSPARSE:
       //TODO: add these test failure cases for cusparse too.
       algo = "SPGEMM_CUSPARSE";
-#ifndef KERNELS_HAVE_CUSPARSE
+#if !defined(KERNELS_HAVE_CUSPARSE) && !defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE)
```
[the remark this answers is not in the document]
Comment (ndellingwood) [reply unit_test/sparse/Test_Sparse_spgemm.hpp]: Not yet, trying to get the Trilinos build on White going.
Are there extra settings to use cusparse like with mkl? For example, adding an extra cxxflag, setting an a path variable, etc? I haven't built kokkos-kernels with cusparse before

[the remark this answers is not in the document]
Comment (kyungjoo-kim) [reply unit_test/sparse/Test_Sparse_spgemm.hpp]: I am not sure. One problematic thing is that "cmake configure" may not match to "make configure". We have two build systems and cmake configure is not tested nor updated in the kokkoskernels level. cusparse or cublas should be ON by default when you test on cuda machines; however, it is possible that some macro variables may not set correctly. Let's wait until your test on white is done.

**Comment by srajama1** [comment]
Please revert after the original issue is fixed.

**Comment by ndellingwood** [comment]
Related to issue #296

**Pull request merged**